create types for seats within cabin

uwe-salomon opened the pull request.

matthias-heinisch commented on types/csmim.obj.seat.1.yaml:
```diff
@@ -0,0 +1,237 @@
+  # - csmim.obj.fault-reporter.1  # Does not exist yet
+description: >
+  Class for the overall status of a seat within the cabin. The seat could be any
+  type of seat. Register at path crew/seats/<row>/<seat>
```
why crew? Aren't there passenger seats in `passenger/seat` and maybe cabin attendant seats in `crew/seat`?
With this commit, there will be a link in the `path` folder, right? note: I would appreciate ATA chapter descriptions to be used (`furnishing/passenger/seat/1/a/`)

uwe-salomon replied:
`crew` is the (security) domain.

However, you are right that we need a path structure for cabin attendant seats, too. Maybe like so?

- Passenger seats at `crew/furnishing/seats/<row>/<column>`
- Cabin attendant seats at `crew/furnishing/crew-seats/<id>`

uwe-salomon replied:
On a related note: In the best practices, we advised to "consider using the ATA subchapter as third path element". In this case, it would be "passenger compartment equipment" for ATA-20. Not sure if this improves the situation here. In particular, shortening the subchapter to `passenger` or `pax` creates more ambiguity because cabin attendant seats are also located in the passenger compartment...

uwe-salomon replied:
As per Patrick's response today, and with Matthias' proposal for domain names, the topic would become `airline/furnishing/seats/<row>/<seat>`.

PatrickSchmehl replied:
airline/furnishing/seats/<row>/<seat> is acceptable for the seat group.

matthias-heinisch commented on types/csmim.obj.seat.1.yaml:
```diff
@@ -0,0 +1,237 @@
+      aircraft cabin, e.g. 11.
+
+resources:
+  - id: seat_position
```
to reduce redundancy, we could shorten this into `position` (the topic includes the `seat` keyword already)

PatrickSchmehl replied:
We have discussed the naming within the seat group. Idea is, that clear naming, even if it doubles information (seat is already in the topic) is preferred for a better understanding. Therefore we want to keep the naming.

matthias-heinisch commented on types/csmim.obj.seat.1.yaml:
```diff
@@ -0,0 +1,237 @@
+        key: 3
+        description: Obstruction event on divider has occurred
+
+  - id: motion_cmd
```
we could shorten this into `motion`, as `mode: x` adds already the `command` keyword

uwe-salomon replied:
I'd propose `move`. Verbs for executable commands?

PatrickSchmehl replied:
We have discussed the naming within the seat group. Idea is, that clear naming, even if it doubles information (mode defines the type) is preferred for a better understanding. Therefore we want to keep the naming.

matthias-heinisch commented on types/csmim.obj.seat.1.yaml:
```diff
@@ -0,0 +1,237 @@
+        type: enum
+        description: Defines how the seat actuation system reacts on the command
+        enum-values:
+          - name: cyclic
```
I would have expected two commands (`move` and `move_to`), but I see the advantage here to prevent contradicting orders (at least as long as there is only one commander). Let's discuss, if a different communication pattern could be advantageous.

PatrickSchmehl replied:
We want to have one command, where all paremeters for the movement is set.

matthias-heinisch commented on types/csmim.obj.seat.1.yaml:
```diff
@@ -0,0 +1,237 @@
+          - name: righthand_down
+            key: 21
+            description: Move right-hand side down, e.g. arm rest
+      - key: source
```
this is non-trusted and I consider it dangerous to offer. Let's see if a different communication pattern can provide reliable information.

DennisWhetten-Boeing replied:
I agree. Perhaps adding a way to look up something (group?) by the username instead?

PatrickSchmehl replied:
I didn't get the question. More explenation is needed.

matthias-heinisch replied:
Currently, the CSMIM protocol allows for two different methods to differentiate senders: a) each command includes an anonymous sender hash, so individual responses are possible. Possibly a lookup-service could provide more information, if this is needed b) instead of using commands, a seat commander could publish the desired behavior into a well-known data topic. As data topics include the path to the sender, different seat commanders can be differentiated and prioritized. This has the benefit of not requiring access control.

uwe-salomon replied:
As Patrick specified in another comment, the "source" is just for logging purposes. I changed the description accordingly. I would still prefer that this parameter be removed completely, to prevent misuse. If you decide to use an executable resource, CSMIM provides no way to identify the sender of the command, apart from an anonymous sender hash. Therefore, you must specify the seat behavior without relying on command source information.

matthias-heinisch commented on types/csmim.obj.seat.1.yaml:
```diff
@@ -0,0 +1,237 @@
+    type: enum
+    description: The ast safety-related event on seat / suit level
+    enum-values:
+      - name: none
```
`none` and `undefined` could be confused. maybe call it `empty`, `free`, or `available`?

uwe-salomon replied:
The last safety-related event: `none`. I find that good to understand.

But what is `undefined`? What is an undefined safety-related event?

matthias-heinisch commented on types/csmim.obj.seat.actuator.1.yaml:
```diff
@@ -0,0 +1,52 @@
+description: >
+  A class for seat actuators in the cabin. Specific data from the manufacturer
+  must be added to derived classes. Register at path
+  crew/seats/<row>/<seat>/actuators/<actuator_id>
```
see comment in `obj.seat.1`

PatrickSchmehl replied:
Tpoic structure needs adapted according to seat class.

matthias-heinisch reviewed (commented):
neat! thank you for your good contribution. I am looking forward to dive into the discussions.

uwe-salomon commented on types/csmim.obj.sas-operational-status.1.yaml:
```diff
@@ -0,0 +1,45 @@
+%YAML 1.2
+---
+id: csmim.obj.sas-operational-status.1
```
`csmim.obj.seat.operational-status.1`
SAS for seat actuation system is cryptic, while "seat" can be understood and groups the types nicely together.

uwe-salomon replied:
Ok. I think it is still important to group our type IDs, because we will have a lot of them in the future. If you want to keep "sas", the type ID would become `csmim.obj.seat.sas-operational-status.1`.

uwe-salomon commented on types/csmim.obj.seat.1.yaml:
```diff
@@ -0,0 +1,237 @@
+          for TTL is indicated via additional resource safe_for_ttl
+      - name: dine
+        key: 3
+        description: The seat is in dine position
```
Many of these descriptions provide no additional information and should be removed.

PatrickSchmehl replied:
You are right, Some comments could be removed. Some comments are help full. Question: Should we mix enumerations with descriptions for some values and some without?

uwe-salomon commented on types/csmim.obj.seat.1.yaml:
```diff
@@ -0,0 +1,237 @@
+      - name: ttl
+        key: 1
+        description: The seat is in TTL position
+      - name: ttl2
```
Are ttl2 and relax2 also displayed on a screen as "TTL 2" and "Relax 2"?

PatrickSchmehl replied:
They are sometimes displayed as TTL Upright,..... So the display naming is often different to the naming. Relax is sometimes also called zero G,......

uwe-salomon commented on types/csmim.obj.seat.1.yaml:
```diff
@@ -0,0 +1,237 @@
+        description: The seat is in TTL position
+      - name: ttl2
+        key: 2
+        description: The seat has more than one TTL position. The status of safe
+          for TTL is indicated via additional resource safe_for_ttl
```
Important information. Should be moved to resource description and generalized.

uwe-salomon commented on types/csmim.obj.seat.1.yaml:
```diff
@@ -0,0 +1,237 @@
+    type: void
+    optional: true
+    description: >
+      Motion command to change seat position. Only one command at the same time
+      is allowed. In case, that a dedicated motion command is not supported, a
+      NAK message shall be sent. The enumeration may be extended in the future.
```
1. "Only one command allowed" conflicts with the "source" entry below. I would propose to remove the "source", and reword this paragraph to specify seat behavior when conflicting commands are received in short succession. You cannot just opt out from problems. The system integrator will then decide whether the behavior is acceptable or whether preventive measures have to be taken.

2. If a motion command is not supported, the response should contain an appropriate error response code, e.g. 400 Bad Request or 501 Not Implemented.

3. If these enumerations are extended in the future: How could a CSMIM client, in particular in the more distant future, obtain a list of valid seat positions and valid move commands?

PatrickSchmehl replied:
Answers:
1.	Idea is to have a possibility to distinguish between different senders of the command if needed and for logging purposes. This field will not solve any certification related issues and is seen more as an additional information.
2.	Is there a list of fault colds available and how do we they specify?
3.	Good question.

uwe-salomon replied:
There is a list of fault codes in the CSMIM standard. I added a description to that avail, to be reviewed by you.

I also added a paragraph to the source parameter, even though I would still prefer to remove that parameter completely.

uwe-salomon commented on types/csmim.obj.seat.1.yaml:
```diff
@@ -0,0 +1,237 @@
+  - csmim.obj.sas-operational-status.1
+  - csmim.obj.seat.occupied-status.1
+  # - csmim.obj.fault-reporter.1  # Does not exist yet
+description: >
```
Should also specify access control for writeable and executable resources. The seats must set at least the object-wide `rmod` property, in order to prevent anybody on the network from sending commands to seats. Please devise an appropriate access control scheme and specify the required CSMIM roles here.

uwe-salomon commented on types/csmim.obj.seat.1.yaml:
```diff
@@ -0,0 +1,237 @@
+            description: Motion is for the complete seat
+          - name: leg_rest
+            key: 2
+            description: Leg rest of the seat
```
Again, many descriptions could be removed because they do not add information.

uwe-salomon commented on types/csmim.obj.seat.1.yaml:
```diff
@@ -0,0 +1,237 @@
+      - key: direction
+        type: enum
+        description: Direction of the motion. Note that not all directions are
+          possible with each part. In case the motion is not supported, a NAK
+          response will be sent.
```
Again, specify CSMIM response code. Note that, here, 400 could be used to respond to invalid combinations (e.g. leg rest clockwise), while 501 could be used to respond to unavailable motions (e.g. leg rest if there is no leg rest).

uwe-salomon commented on types/csmim.obj.seat.actuator.1.yaml:
```diff
@@ -0,0 +1,52 @@
+    description: ID of the actuator
+
+resources:
+  - id: obstruction_flag
```
Propose `obstruction` or to `obstructed` or `is_obstructed`.
See best practices / identifiers for boolean resources.

PatrickSchmehl replied:
Correct should be changed to: `is_obstructed`

uwe-salomon commented on types/csmim.obj.seat.occupied-status.1.yaml:
```diff
@@ -0,0 +1,26 @@
+attributes: []
+
+resources:
+  - id: occupied_status
```
Propose just `occupied` or `occupiedness`?

PatrickSchmehl replied:
`occupied`

uwe-salomon commented on types/csmim.obj.seat.occupied-status.1.yaml:
```diff
@@ -0,0 +1,26 @@
+      - name: baggage
+        key: 2
+        description: Baggage is detected on seat
+      - name: unknown
+        key: 3
+        description: Signal from sensor is indefinite
```
This value would signal to a receiver "there is an error here but it's the fault of the sensor not of the seat controller". Whereas, if the seat controller itself failed, most likely the resource would cease to be published. A receiver has to check for that case anyways. I propose to remove this enum value, and instead cease to publish the resource value in case the sensor is defect or whatever. Then use the fault reporting mechanism to communicate the exact nature of the problem to the system.

matthias-heinisch commented:
I would also recommend to split the seat into a status-reporting-seat and a remote-controlled-seat on top of that. This would allow us to introduce the first iteration without probably complex security reviews that are necessary for remote control functions.

PatrickSchmehl commented:
We discussed the topic within the seat group, if we should separate the motion command from the seat class due to possible certification discussions. We belief, that the motion command will be used more and more in the future it will be a standard. Therefore, we push back the idea of splitting the seat class in two.

uwe-salomon commented on types/csmim.obj.seat.1.yaml:
```diff
@@ -0,0 +1,199 @@
+          - name: cleaning
+            key: 9
+          - name: up
+            key: 100
+          - name: down
+            key: 101
+          - name: in
+            key: 102
+          - name: out
+            key: 103
+          - name: forward
+            key: 104
+          - name: backward
+            key: 105
+          - name: rotate_clockwise
+            key: 106
+          - name: rotate_counterclockwise
+            key: 107
+          - name: lefthand_up
+            key: 108
+          - name: lefthand_down
+            key: 109
+          - name: righthand_up
+            key: 110
+          - name: righthand_down
+            key: 111
```
Note that I increased the enum keys here. This makes room for extending the list of pre-defined positions here and in the "seat_position" resource, and keep the keys identical in both cases.

matthias-heinisch commented on types/csmim.obj.seat.1.yaml:
```diff
@@ -0,0 +1,199 @@
+attributes:
+  - key: row
+    description: Row number of the installation location of the seat within the
+      aircraft cabin, e.g. "11".
```
I suppose this is the seat row label? so for example a 15 for pyhsical seatrow 14 in countries where no seatrow is labelled with 13, correct?

uwe-salomon commented on types/csmim.obj.seat.1.yaml:
```diff
@@ -0,0 +1,199 @@
+        enum-values:
+          - name: cyclic
+            key: 1
+            description: Commands needs to be sent cyclically, otherwise the
```
Note that legacy systems seem to use a 500ms cycle.

uwe-salomon commented on types/csmim.obj.seat.1.yaml:
```diff
@@ -0,0 +1,199 @@
+      - name: cleaning
+        key: 9
+
+  - id: safe_for_ttl
+    mode: r
+    type: bool
+    description: >
+      All criteria for safe to flight on seat/suit level are fulfilled. E.g. the
+      seat is in TTL position and all furniture is correctly stored.
```
Discussion in Swindon showed that a ready may confuse on which level the resources apply. In particular, the seat "position" and "motion" command should be applied to the physical seat only (not the suite), while the "safe_for_ttl" accumulates the status of the whole suite including all furniture. We should check the descriptions to make this distinction clear.

uwe-salomon commented on types/csmim.obj.seat.1.yaml:
```diff
@@ -0,0 +1,199 @@
+            key: 8
+          - name: ottoman
+            key: 9
+          - name: privacy_divider
+            key: 10
```
The motion command allows to command the privacy divider. However, there is no interface to read the divider's position. This could be ok. An alternative is to create a "privacy_divider" object type and register it under seat/divider, with resources like seat/divider/position and seat/divider/motion.

uwe-salomon commented on types/csmim.obj.seat.1.yaml:
```diff
@@ -0,0 +1,199 @@
+      - name: divider_obstruction
+        key: 3
+
+  - id: motion
+    mode: x
```
Patrick replied that the motion command is intended to be a superset of all possible movement command options. The details will be worked out for each installation between the airline, the seat supplier and the IFE supplier. This is also how the IFE "knows" what can be commanded actually. Patrick said this process is normal and not a problem.

uwe-salomon commented:
I just rebased to the new "main" branch to take advantage of the bite-reporter type.
And I changed the last commit from Dennis to use the new type ID.

PatrickSchmehl commented:
The enum change is good for us.

Best Regards

Patrick



--​‍​
Mit freundlichen Grüßen / Best regards	
	
i.A. Patrick Schmehl	
	
Bühler Motor Aviation GmbH	
Project & Product Management	
Product Manager	
	
Tel.: +49 (7556) 9225 39 | Mobile: +4915222967778  | E-Mail: ***@***.***	
Hallendorfer Str. 11 | 88690 Uhldingen-Mühlhofen | Deutschland | Web: www.buehlermotor.de 	
Reg.Ger. HRB 701496 Amtsgericht Freiburg | Geschäftsführer: Mark Furtwängler, Andreas Ulmer | USt-IdNr.: DE 256981605

Von: matthias ***@***.***>
Gesendet: Dienstag, 16. Dezember 2025 13:20
An: ARINC-IA/CSMIM ***@***.***>
Cc: Patrick Schmehl ***@***.***>; Comment ***@***.***>
Betreff: Re: [ARINC-IA/CSMIM] create types for seats within cabin (PR #15)


@matthias-heinisch approved this pull request.

very nice, thank you for your good work!
If the latest change (enum update) is good for the seat group, we can merge it to main.

—
Reply to this email directly, view it on GitHub<https://atpscan.global.hornetsecurity.com?d=ssh3hAxkXaA5H4Upvpb9ZGdSxX0pmUBs2EuRt9ebxu8YNYBm9PiIvc49pdot-Z0Z&f=dTZ--DFtNpz0hQJquhwiBgBFnZ4IJ5pF6axJ8Q9Gz9PCxj5RqEo4WFFLDohqZAZm&i=&k=gXnD&m=MjdmdDQ-u5MyMNXdY8Hr91Rf0I6lHKEP-oJJ_lXCa49R9_Kjo5Bt56IdZas4WQUMwTuLlgmr4neGEsAfB6ngmhQw5OQ8ACA-yfVnv356K_TGjMSA9MWgPDBD6169kHkK&n=PA43QAF0Xsfblr0-re-wSFClrcmkrGWaX7aV-1gE45_WBjeJr4u00LuJ4-5uun9P&r=JsELiFsO7xSzfHR8R4N7dPV76lwkBDnNwa5WZsxR2jnDvGhzY7dxhSxo58NvOfy6pR3D_m_-GG-yHn0GDrTqDA&s=1c73e71ed98e0f521ffe1418cb26e223beaedaf39b9970c44e15cd9199fdadb0&u=https%3A%2F%2Fgithub.com%2FARINC-IA%2FCSMIM%2Fpull%2F15%23pullrequestreview-3582799731>, or unsubscribe<https://atpscan.global.hornetsecurity.com?d=ICN4iBZnxE0n1dm6-Mq0UZodOro14t8LDJUO0nPT8H8vAejKZcAf0fkQ31wM9JF3&f=3U5bPv1AVBvw6Xy-JN9NsWvJ4wiM4N5KCfoet0u3yFp9Lwx9HlLwa1BgX7xf1mbw&i=&k=4gWG&m=SRK3C-gRnSaKc5-BTbq_D2nFa9BIZ-xKa3seLgivnnTkbMSBYy2rSpand74LVIwydU-dTgPSvY9TMKhLhgFjvLuejp1n4TVimPLqRI1DGCDYroDMBNGKVWmdNosFtAeJ&n=11oohB4lu46ginX5VbDjv1TGKeimfWHJciiidrSDuiSdVIabRW7n7iFF1yA8pjZl&r=BRcRTX9NGhtB8TGTlO4Svy2hZSB_vpEIrShlFoztj9jFs2mzo6zU2BTE5VqyV570obYOd46y4UPhnUN1_rOblw&s=e4b352cdc48fd1af9a2e354c8fba5c11dca0060e510299fca3e4023c6e3401f5&u=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FBKDK5NJWLAXFKZ2UW2M6BWL4B72H5AVCNFSM6AAAAABWHJPEMKVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZTKOBSG44TSNZTGE>.
You are receiving this because you commented.Message ID: ***@***.******@***.***>>
